File: core/src/com/xenoage/zong/core/music/direction/Segno.java
package com.xenoage.zong.core.music.direction;

import lombok.Data;
import lombok.EqualsAndHashCode;

import com.xenoage.zong.core.header.ColumnHeader;
import com.xenoage.zong.core.music.MusicElementType;
import com.xenoage.zong.core.position.MP;


/**
 * Class for a segno sign.
 * 
 * It must always be written to the {@link ColumnHeader}.
 *
 * @author Andreas Wenger
 */
@Data @EqualsAndHashCode(callSuper=false)
public final class Segno
	extends Direction
	implements NavigationSign {

	@Override public MusicElementType getMusicElementType() {
		return MusicElementType.Segno;
	}
	
	@Override public MP getMP() {
		return MP.getMPFromParent(this);
	}
	
}

File: core/src/com/xenoage/zong/core/music/direction/Segno.kt
package com.xenoage.zong.core.music.direction

import com.xenoage.zong.core.header.ColumnHeader
import com.xenoage.zong.core.music.format.Positioning


/**
 * Class for a segno sign.
 */
class Segno : NavigationSign {

	/** True, iff repeats should be played after jumping back (con repetizione or senza repetizione).  */
	var isWithRepeats: Boolean = true

	override var positioning: Positioning? = null

	override var parent: ColumnHeader? = null

}
